tests/keras/engine/test_topology.py

```diff
@@ -550,6 +582,59 @@ def test_load_layers():
 
 
 @keras_test
+def test_preprocess_weights_for_loading():
+    from keras.engine.topology import preprocess_weights_for_loading
+
+    def list_allclose(x_list, y_list):
+        return all([np.allclose(x, y, 1e-5)
+                    for (x, y) in zip(x_list, y_list)])
+
+    def convert_weights(layer, weights):
+        if layer.__class__.__name__ == 'GRU':
+            W = [np.split(w, 3, axis=-1) for w in weights]
+            return sum(map(list, zip(*W)), [])
+        elif layer.__class__.__name__ in ('LSTM', 'ConvLSTM2D'):
+            W = [np.split(w, 4, axis=-1) for w in weights]
+            for w in W:
+                w[2], w[1] = w[1], w[2]
+            return sum(map(list, zip(*W)), [])
+        elif layer.__class__.__name__ == 'Conv2DTranspose':
+            return [np.transpose(weights[0], (2, 3, 0, 1)), weights[1]]
+        return weights
+
+    test_layers = [
```

Comment:
This approach is fine by me, but you could use https://docs.pytest.org/en/latest/parametrize.html to move the for loop outside the test. Totally up to you, just something I've seen done.